core/src/test/scripts/unit_tests_validate_data.py

```diff
@@ -1012,7 +1013,198 @@ def test_missing_row_gsva(self):
+                                    validateData.CNADiscreteLongValidator,
+                                    extra_meta_fields={'namespaces': 'MyNamespace'})
+        # expecting only status messages about the file being validated
+        self.assertEqual(5, len(record_list))
+        for record in record_list:
+            self.assertLessEqual(record.levelno, logging.INFO)
+
+    def test_entrez_only(self):
+        """Test when a file has an Entrez ID column but none for Hugo names."""
+        record_list = self.validate('data_cna_long_genecol_presence_entrez_only.txt',
+                                    validateData.CNADiscreteLongValidator,
+                                    extra_meta_fields={'namespaces': 'MyNamespace'})
+        # expecting only status messages about the file being validated
+        self.assertEqual(5, len(record_list))
+        for record in record_list:
+            self.assertLessEqual(record.levelno, logging.INFO)
+
+    def test_neither_genic_columns(self):
+        """Test when a file lacks the Entrez ID and Hugo name columns."""
+        record_list = self.validate('data_cna_long_genecol_presence_neither_genic.txt',
+                                    validateData.CNADiscreteLongValidator,
+                                    extra_meta_fields={'namespaces': 'MyNamespace'})
+        # two errors after the info: the first makes the file unparsable
+        self.assertEqual(3, len(record_list))
+        for record in record_list[1:]:
+            self.assertEqual(logging.ERROR, record.levelno)
+        self.assertEqual(1, record_list[1].line_number)
+
+    # Tests validating gene-wise files with different combinations of gene id columns,
+    # now with invalid Entrez ID and/or Hugo names """
+
+    def test_missing_sample_column(self):
+        """Test when a file do not have the Sample_Id column."""
+        record_list = self.validate('data_cna_long_genecol_no_sample_id.txt',
+                                    validateData.CNADiscreteLongValidator,
+                                    extra_meta_fields={'namespaces': 'MyNamespace'})
+        # two errors after the info: the first makes the file unparsable
+        self.assertEqual(3, len(record_list))
+        for record in record_list[1:]:
+            self.assertEqual(logging.ERROR, record.levelno)
+        self.assertEqual(1, record_list[1].line_number)
+    def test_missing_value_column(self):
+        """Test when a file do not have the Value column."""
+        record_list = self.validate('data_cna_long_genecol_no_value.txt',
+                                    validateData.CNADiscreteLongValidator,
+                                    extra_meta_fields={'namespaces': 'MyNamespace'})
+        # two errors after the info: the first makes the file unparsable
+        self.assertEqual(3, len(record_list))
+        for record in record_list[1:]:
+            self.assertEqual(logging.ERROR, record.levelno)
```

Comment:
I can see that there are some tests that check error messages. It's better to apply it to all tests that expect error-level logs.